cmd/crank/beta/render/cmd.go

```diff
@@ -28,24 +28,64 @@ import (
 
 	"github.com/crossplane/crossplane-runtime/pkg/errors"
 	"github.com/crossplane/crossplane-runtime/pkg/logging"
-	"github.com/crossplane/crossplane-runtime/pkg/resource/unstructured/composed"
 
 	v1 "github.com/crossplane/crossplane/apis/apiextensions/v1"
 )
 
 // Cmd arguments and flags for render subcommand.
 type Cmd struct {
-	Timeout time.Duration `help:"How long to run before timing out." default:"1m"`
+	// Arguments.
+	CompositeResource string `arg:"" type:"existingfile" help:"A YAML file specifying the composite resource (XR) to render."`
+	Composition       string `arg:"" type:"existingfile" help:"A YAML file specifying the Composition to use to render the XR. Must be mode: Pipeline."`
+	Functions         string `arg:"" type:"path" help:"A YAML file or directory of YAML files specifying the Composition Functions to use to render the XR."`
+
+	// Flags. Keep them in alphabetical order.
+	ContextFiles           map[string]string `mapsep:"," help:"Comma-separated context key value pairs to pass to the Function pipeline. Values must be files containing JSON."`
+	ContextValues          map[string]string `mapsep:"," help:"Comma-separated context key value pairs to pass to the Function pipeline. Values must be JSON. Keys take precedence over --context-files."`
+	IncludeFunctionResults bool              `short:"r" help:"Include informational and warning messages from Functions in the rendered output as resources of kind: Result."`
+	ObservedResources      string            `short:"o" placeholder:"PATH" type:"path" help:"A YAML file or directory of YAML files specifying the observed state of composed resources."`
+	Timeout                time.Duration     `help:"How long to run before timing out." default:"1m"`
+}
```

Comment:
This renders as:

```
$ go run cmd/crank/main.go beta render --help
Usage: crossplane beta render <composite-resource> <composition> <functions>

Render a composite resource (XR).

Crossplane uses a Composition to produce a set of composed resources for a given
XR. This command shows you what composed resources Crossplane would create by
printing them to stdout. It also prints any changes that would be made to the
status of the XR.

This command doesn't talk to Crossplane. Instead it runs the Composition
Function pipeline specified by the Composition locally, and uses that to render
the XR. It only supports Compositions that use the Pipeline mode.

Composition Functions are pulled and run using Docker by default. You can add
the following annotations to each Function to change how they're run:

      render.crossplane.io/runtime: "Development"

        Connect to a Function that is already running, instead of using Docker. This
        is useful to develop and debug new Functions. The Function must be listening
        at localhost:9443 and running with the --insecure flag.

      render.crossplane.io/runtime-development-target: "dns:///example.org:7443"

        Connect to a Function running somewhere other than localhost:9443. The
        target uses gRPC target syntax.

      render.crossplane.io/runtime-docker-cleanup: "Orphan"

        Don't stop the Function's Docker container after rendering.

      render.crossplane.io/runtime-docker-pull-policy: "Always"

        Always pull the Function's OCI image, even if it already exists locally.
        Other supported values are Never, or IfNotPresent.

Use the standard DOCKER_HOST, DOCKER_API_VERSION, DOCKER_CERT_PATH, and
DOCKER_TLS_VERIFY environment variables to configure how this command connects
to the Docker daemon.

Arguments:
  <composite-resource>    A YAML file specifying the composite resource (XR) to
                          render.
  <composition>           A YAML file specifying the Composition to use to
                          render the XR. Must be mode: Pipeline.
  <functions>             A YAML file or directory of YAML files specifying the
                          Composition Functions to use to render the XR.

Flags:
  -h, --help                       Show context-sensitive help.
      --verbose                    Print verbose logging statements.
  -v, --version                    Print version and quit.

      --context-files=KEY=VALUE,...
                                   Comma-separated context key value pairs to
                                   pass to the Function pipeline. Values must be
                                   files containing JSON.
      --context-values=KEY=VALUE,...
                                   Comma-separated context key value pairs
                                   to pass to the Function pipeline. Values
                                   must be JSON. Keys take precedence over
                                   --context-files.
  -r, --include-function-results
                                   Include informational and warning messages
                                   from Functions in the rendered output as
                                   resources of kind: Result.
  -o, --observed-resources=PATH    A YAML file or directory of YAML files
                                   specifying the observed state of composed
                                   resources.
      --timeout=1m                 How long to run before timing out.
```

I don't love how long the help block is in this one. Feels like we might struggle to keep it updated. Not sure if I have any better ideas though. Perhaps we could link to a documentation page dedicated to this command?